napari/_qt/qt_event_loop.py

```diff
@@ -1,38 +1,123 @@
+        NAPARI_ICON_PATH
+    org_name : str, optional
+        Set organization name (if creating for the first time), by default
+        'napari'
+    org_domain : str, optional
+        Set organization domain (if creating for the first time), by default
+        'napari.org'
+    app_id : str, optional
+        Set organization domain (if creating for the first time).  Will be
+        passed to set_app_id (which may also be called independently), by
+        default NAPARI_APP_ID
+
+    Returns
+    -------
+    QApplication
+        [description]
 
     Notes
     -----
-    Substitute QApplicationWithTracing when using perfmon.
+    Substitutes QApplicationWithTracing when the NAPARI_PERFMON env variable
+    is set.
 
-    With IPython/Jupyter we call convert_app_for_tracing() which deletes
-    the QApplication and creates a new one. However here with gui_qt we
-    need to create the correct QApplication up front, or we will crash.
-    We'll crash because we'd be deleting the QApplication after we created
-    QWidgets with it, such as we do for the splash screen.
+    If the QApplication already exists, we call convert_app_for_tracing() which
+    deletes the QApplication and creates a new one. However here with get_app
+    we need to create the correct QApplication up front, or we will crash
+    because we'd be deleting the QApplication after we created QWidgets with
+    it, such as we do for the splash screen.
     """
-    if perf_config and perf_config.trace_qt_events:
-        from .perf.qt_event_tracing import QApplicationWithTracing
 
-        return QApplicationWithTracing(argv)
+    app = QApplication.instance()
+    if app:
+        if perf_config and perf_config.trace_qt_events:
+            from .perf.qt_event_tracing import convert_app_for_tracing
+
+            # no-op if app is already a QApplicationWithTracing
+            app = convert_app_for_tracing(app)
+        app._existed = True
     else:
-        return QApplication(argv)
+        # automatically determine monitor DPI.
+        # Note: this MUST be set before the QApplication is instantiated
+        QApplication.setAttribute(Qt.AA_EnableHighDpiScaling)
+
+        if perf_config and perf_config.trace_qt_events:
+            from .perf.qt_event_tracing import QApplicationWithTracing
+
+            app = QApplicationWithTracing(sys.argv)
+        else:
+            app = QApplication(sys.argv)
+
+        # if this is the first time the Qt app is being instantiated, we set
+        # the name, so that we know whether to raise_ in Window.show()
+
+        app.setApplicationName(app_name)
+        app.setApplicationVersion(app_version)
+        app.setOrganizationName(org_name)
+        app.setOrganizationDomain(org_domain)
+        app.setWindowIcon(QIcon(icon))
+
+    if perf_config and not perf_config.patched:
+        # Will patch based on config file.
+        perf_config.patch_callables()
+
+    set_app_id(app_id)
```

Comment:
I would prefer it to be called only when the app is created as well, I only put it there for your convenience, since it's more similar to the behavior in master.  I'll make it stricter